internal/repository/gorm/tag.go

```diff
@@ -0,0 +1,115 @@
+package gorm
+
+import (
+	"fmt"
+
+	"github.com/porter-dev/porter/internal/models"
+	"github.com/porter-dev/porter/internal/repository"
+	"gorm.io/gorm"
+)
+
+// AllowlistRepository uses gorm.DB for querying the database
+type TagRepository struct {
+	db *gorm.DB
+}
+
+// NewAllowlistRepository returns a AllowListRepository which uses
+// gorm.DB for querying the database.
+func NewTagRepository(db *gorm.DB) repository.TagRepository {
+	return &TagRepository{db}
+}
+
+func (repo *TagRepository) CreateTag(tag *models.Tag) (*models.Tag, error) {
+	existingTag, _ := repo.ReadTagByNameAndProjectId(tag.Name, tag.ProjectID)
+
+	if existingTag != nil {
+		return nil, fmt.Errorf("tag already exists")
+	}
+
+	if err := repo.db.Create(tag).Error; err != nil {
+		return nil, err
+	}
+	return tag, nil
+}
+
+func (repo *TagRepository) LinkTagsToRelease(tags []string, release *models.Release) ([]*models.Tag, error) {
+	populatedTags := make([]*models.Tag, 0)
+	err := repo.db.Model(&models.Tag{}).Where("name IN ?", tags).Where("project_id = ?", release.ProjectID).Find(&populatedTags).Error
```

Comment:
```suggestion
	err := repo.db.Where("name IN ?", tags).Where("project_id = ?", release.ProjectID).Find(&populatedTags).Error
```

I could be wrong but I don't think you need the `Model` section since the `Find` statement defines the table for the resource you're looking for.

Reply:
Nice! Didn't knew this one :raised_hands: